New resources: azurerm_eventhub_namespace_dedicated and azurerm_eventhub_dedicated (#7347)
terraform-providers/terraform-provider-azurerm: website/docs/r/eventhub_dedicated.html.markdown

---
subcategory: "Messaging"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_eventhub_dedicated"
description: |-
  Manages a Event Hubs as a nested resource within an Event Hubs namespace on an Event Hubs Cluster.
---

# azurerm_eventhub_dedicated

Manages a Event Hubs as a nested resource within an Event Hubs namespace on an Event Hubs Cluster.

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "resourceGroup1"
  location = "West US"
}

resource "azurerm_eventhub_cluster" "example" {
  name                = "eventhubclusterexample"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location
  sku_name            = "Dedicated_1"
}

resource "azurerm_eventhub_namespace_dedicated" "example" {
  name                = "acceptanceTestEventHubNamespace"
  cluster_id          = azurerm_eventhub_cluster.example.id
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

> resource "azurerm_eventhub" "example" {
>   name                = "acceptanceTestEventHub"
>   namespace_name      = azurerm_eventhub_namespace.example.name
>   resource_group_name = azurerm_resource_group.example.name
>   partition_count     = 2
>   message_retention   = 1
> }

Reviewer: Wrong resource for the example here?
```suggestion
resource "azurerm_eventhub_dedicated" "example" {
  name                = "acceptanceTestEventHub"
  namespace_name      = azurerm_eventhub_namespace_dedicated.example.name
  resource_group_name = azurerm_resource_group.example.name
  partition_count     = 2
  message_retention   = 1
}
```

```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the EventHub Namespace resource. Changing this forces a new resource to be created.

* `namespace_name` - (Required) Specifies the name of the EventHub Namespace. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the resource group in which the EventHub's parent Namespace exists. Changing this forces a new resource to be created.

* `partition_count` - (Required) Specifies the current number of shards on the Event Hub. Needs to be between 1 and 1024. Changing this forces a new resource to be created.

> * `message_retention` - (Required) Specifies the number of days to retain the events for this Event Hub. Needs to be between 1 and 90 days; or 1 day when using a Basic SKU for the parent EventHub Namespace.

Reviewer: ```suggestion
* `message_retention` - (Required) Specifies the number of days to retain the events for this Event Hub. Needs to be between `1` and `90`. Only a value of `1` is allowed when using a Basic SKU for the parent EventHub Namespace.
```

* `destination` - (Required) A `destination` block as defined below.

A `destination` block supports the following:

* `name` - (Required) The Name of the Destination where the capture should take place. At this time the only supported value is `EventHubArchive.AzureBlockBlob`.

-> At this time it's only possible to Capture EventHub messages to Blob Storage. There's [a Feature Request for the Azure SDK to add support for Capturing messages to Azure Data Lake here](https://github.com/Azure/azure-rest-api-specs/issues/2255).

* `archive_name_format` - The Blob naming convention for archiving. e.g. `{Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}`. Here all the parameters (Namespace,EventHub .. etc) are mandatory irrespective of order

* `blob_container_name` - (Required) The name of the Container within the Blob Storage Account where messages should be archived.

* `storage_account_id` - (Required) The ID of the Blob Storage Account where messages should be archived.

## Attributes Reference

The following attributes are exported:

* `id` - The ID of the EventHub.

* `partition_ids` - The identifiers for partitions created for Event Hubs.

> ## Timeouts
> 
> 
> 
> The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

Reviewer: additional whitespace
```suggestion
## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:
```
